### PR TITLE
remove temporary files when copy

### PR DESCRIPTION
**Pull request by HirazawaUi**
[NO NEW TESTS NEEDED]

Fixes: #19496

#### Does this PR introduce a user-facing change?

```release-note
podman cp via STDIN will delete the temporary files
```

### CODE REVIEW

File: cmd/podman/containers/cp.go [hunk 1]
```diff
@@ -382,13 +382,16 @@ func copyToContainer(container string, containerPath string, hostPath string) er
 		if err != nil {
 			return err
 		}
+		defer os.Remove(tmpFile.Name())
+
 		_, err = io.Copy(tmpFile, os.Stdin)
 		if err != nil {
 			return err
 		}
 		if err = tmpFile.Close(); err != nil {
```
Comment (rhatdan) [inline cmd/podman/containers/cp.go:391]: This shoould be deferred also.

Comment (giuseppe) [reply cmd/podman/containers/cp.go:391]: I think it is better if we close immediately after we finish writing to it, but we must make sure it is also closed in the error path of `io.Copy`.

**Review by Luap99 — APPROVED** [review]
LGTM

File: cmd/podman/containers/cp.go (hunk 1 above)
Comment (giuseppe) [reply cmd/podman/containers/cp.go:391]: opened a PR to close the file on `io.Copy` errors: https://github.com/containers/podman/pull/19534

**Pull request merged**